source/beamline/BioXAS/BioXASSSRLMonochromatorRegionControl.h

```diff
@@ -0,0 +1,284 @@
+	virtual bool stop() { return false; }
+
+	// Sets the upper slit motor control.
+	void setUpperSlitControl(AMControl *upperSlit);
+	/// Sets the lower slit motor control.
+	void setLowerSlitControl(AMControl *lowerSlit);
+	/// Sets the slits status control.
+	void setSlitsStatusControl(AMControl *slitsStatus);
+	/// Sets the paddle motor control.
+	void setPaddleControl(AMControl *paddle);
+	/// Sets the paddle status control.
+	void setPaddleStatusControl(AMControl *paddleStatus);
+	/// Sets the key status control.
+	void setKeyStatusControl(AMControl *keyStatus);
+	/// Sets the brake status control.
+	void setBrakeStatusControl(AMControl *brakeStatus);
+	/// Sets the bragg motor control.
+	void setBraggControl(AMControl *bragg);
+	/// Sets the bragg at crystal change position status control.
+	void setBraggAtCrystalChangePositionStatusControl(AMControl *inPosition);
+	/// Sets the crystal change motor control.
+	void setCrystalChangeControl(AMControl *crystalChange);
+	/// Sets the crystal change cw limit status control.
+	void setCrystalChangeCWLimitStatusControl(AMControl *limitStatus);
+	/// Sets the crystal change ccw limit status control.
+	void setCrystalChangeCCWLimitStatusControl(AMControl *limitStatus);
+	/// Sets the region A status control.
+	void setRegionAStatusControl(AMControl *regionStatus);
+	/// Sets the region B status control.
+	void setRegionBStatusControl(AMControl *regionStatus);
+
+protected slots:
+	/// Updates the current value_ and emits the valueChanged() signal.
+	void setValue(int newValue);
+	/// Updates the current setpoint_ and emits the setpointChanged() signal.
+	void setSetpoint(int newSetpoint);
+	/// Updates the current moveInProgress_ value and emits the moveChanged() signal. The moveStarted() signal is also emitted if the change is to true.
+	void setMoveInProgress(bool isMoving);
+
+	/// Handles emitting the appropriate signals when the current step in a move has changed.
+	void onActionStepChanged(int stepIndex);
+	/// Handles updating the region value and emitting the value changed signal.
+	void onRegionControlValueChanged();
+	/// Called when the internal crystal change action has been started. Handles updating the moveInProgress_ member variable and emitting the moveInProgress() signal.
+	void onRegionChangeStarted();
+	/// Called when the internal crystal change action has been cancelled. Handles emitting moveFailed(...) with the WasStoppedFailure code and deleting the action.
+	void onRegionChangeCancelled();
+	/// Called when the internal crystal change action has failed. Handles emitting moveFailed(...) with the OtherFailure code and deleting the action.
+	void onRegionChangeFailed();
+	/// Called when the internal crystal change action has succeeded! Handles emitting moveSucceeded() and deleting the action.
+	void onRegionChangeSucceeded();
+
+protected:
+	/// Returns a new action that performs a crystal change to change the region.
+	AMListAction3* createChangeRegionAction(int newRegion);
```

Comment:
Ah, okay.  And the AMControl * getters are there just so you can retrieve the various mono controls from the beamline?